### PR TITLE
fix: pass updated NODE_OPTIONS to Node.js

### LINKED ISSUE #27551: Packaged apps allow --require in NODE_OPTIONS on Windows
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 9.3.5
* **Operating System:**
  * Windows 10
* **Last Known Working Electron version:**
  * N/A

### Expected Behavior

According to https://www.electronjs.org/docs/api/environment-variables#node_options or https://github.com/electron/electron/issues/26101 or https://github.com/electron/electron/pull/22539/files it seems packaged electron apps (where executable is not named electron.exe) should not allow --require option in NODE_OPTIONS env variable. 

### Actual Behavior

It seems packaged apps such as latest version of Visual Studio Code permit using --require in NODE_OPTIONS and execute the provided script. 

### To Reproduce

Download latest VS Code and run following:
```cmd
echo console.log('Hello world!');process.exit(0); > c:\temp\test.js
set NODE_OPTIONS="--require=c:/temp/test.js"
code.exe
```
VS code will print Hello world! and exit. 

### Screenshots

image
image

### Additional Information

Issue is reproducible also with v11.2.1 downloaded from npmjs.com.

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
Fixes #27551

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes
Notes: Fixed packaged apps allowing `--require` in `NODE_OPTIONS` on Windows.

### CODE REVIEW

**Review by codebytere — APPROVED** [review]

File: shell/common/node_bindings.cc [hunk 1]
```diff
@@ -280,6 +280,10 @@ void SetNodeOptions(base::Environment* env) {
 
     // overwrite new NODE_OPTIONS without unsupported variables
     env->SetVar("NODE_OPTIONS", options);
+    // https://github.com/electron/electron/issues/27551
+#if defined(OS_WIN)
```
Comment (codebytere) [inline shell/common/node_bindings.cc:284]: on further thought - the root issue seems to be 
> Electron is setting updated NODE_OPTIONS using SetEnvironmentVariable and Node is retrieving them via getenv.

I think instead here we should remove this platform specificity and instead also call

```cpp
setenv("NODE_OPTIONS", options);
```

**Comment by deepak1556** [comment]
Can we backport https://github.com/nodejs/node/commit/a16f0f427eb7bc8c31d8f93fd37bc42cdcf594f8 instead which resorts to using `uv_os_getenv` that will use [GetEnvironmentVariableW](https://github.com/libuv/libuv/blob/b201c1a0f0b1ba2365dc285f466ff6fe5307decf/src/win/util.c#L1540) on windows and [getenv](https://github.com/libuv/libuv/blob/b201c1a0f0b1ba2365dc285f466ff6fe5307decf/src/unix/core.c#L1329) on posix

File: shell/common/node_bindings.cc (hunk 1 above)
Comment (miniak) [reply shell/common/node_bindings.cc:279]: We still need to call this version of the API on Windows as it's supposed to be safer than plain `setenv`, according to MSDN:
> Creates, modifies, or removes environment variables. These are versions of _putenv, _wputenv but have security enhancements, as described in Security Features in the CRT.

Comment (miniak) [reply shell/common/node_bindings.cc]: replaced by @deepak1556 suggestion

**Review by deepak1556 — APPROVED** [review]

**Review by zcbenz — APPROVED** [review]

**Review by MarshallOfSound — COMMENTED** [review]
I'm 👍  on this change but given the vaguely security related nature of this bug can we get a test that verifies our `NODE_OPTIONS` patch works?  Happy for that to be in a follow up PR but it seems like something we should definitely catch sooner

**Comment by miniak** [comment]
@MarshallOfSound I will do that in separate PR

**Review by MarshallOfSound — APPROVED** [review]

**Pull request merged**